### DISCUSSION TITLE
DB Connection Panel - Clicking on "Cancel" does not stop the connection attempt

### DISCUSSION THREAD

**Discussion by steromano87** [dbeaver/dbeaver#354] [Q&A] [answered]
If you double-click on a DB conneciton, the Connection dialog appears (unless you have flagged the "Always run in background" tick). During connection attempt, if you press the "Cancel" button, it has no effect and the connection call proceeds without stopping.

![image](https://cloud.githubusercontent.com/assets/8759452/14420032/f8992fd8-ffc9-11e5-9b13-ec20741833e0.png)

Note: the screenshot refers to an Oracle connection, but this bug appears with every connection type.

**Comment by serge-rider** [comment, marked as answer]
Most drivers do not support connection cancellation. They just stuck in socket connection until timeout will happen (btw you can configure connectTimeout in many drivers). Some drivers use NIO and connection can be canceled at any moment - but this is a rare case.
So just can't fix that, this is driver-dependent thing.
